internal/dkg/execution.go

```diff
@@ -146,8 +146,17 @@ func (d *Process) executeAndFinishDKG(ctx context.Context, beaconID string, conf
 		return err
 	}
 
+	// the `Close()` function of the DKG process could close this channel before we write the results of the DKG to it
+	// so let's recover the panic and return an error; it should only happen when the daemon is shutting down anyway
+	defer func() {
+		if r := recover(); r != nil {
```

Comment:
do you not want to vet that the panic was due to a closed channel? i.e. this will catch any panic and report it as sending on a closed channel

Reply:
hmm we could compare its string value, but I don't think it adds much. Alternatively we could just say 'recovered from panic: %v'

Reply:
Wow, I'm surprised there isn't even a constant you can refer to. I think neither option is great tbh, but given the context of the daemon shutting down it might be okay. Is there a mechanic to check if the daemon is shutting down (so that you can re-panic in case some satanic codepath leads us here while the node is running)?

Reply:
I think only the `done` channel we're already checking in the `select`, but that's the timing issue :( an alternative is put a really big gory log-line in here like:
"THIS SHOULD ONLY EVER HAPPEN IN TEST RUNS AND IF IT DOESN'T THEN OH GOD MAN THE HARPOONS AND SOUND THE ALARMS"